src-tauri/src/swarm.rs

```diff
@@ -74,59 +74,88 @@ pub fn is_swarm_mode_running() -> bool {
     return false;
 }
 
+#[tauri::command(async)]
 pub fn create_environment(handle: tauri::AppHandle) -> String {
-    // Get the application data directory
-    let app_data_dir = tauri::api::path::home_dir().expect("🙈 Failed to get app data directory");
-    let app_data_dir = app_data_dir
-        .join(".config/prem");
-    let app_data_dir = app_data_dir
-        .to_str()
-        .expect("🙈 Failed to convert app data dir path to str");
-
-    // Get create env path
-    let binding = handle.path_resolver()
-        .resolve_resource("petals")
-        .expect("🙈 Failed to find `create_env.sh`");
-    let petals_path = binding
-        .to_str()
-        .expect("🙈 Failed to convert petals path to str");
-
-    let python = format!("{app_data_dir}/miniconda/envs/prem_app/bin/python");
-
-    // Set env variables
-    let mut env = HashMap::new();
-    env.insert("PREM_APPDIR".to_string(), app_data_dir.to_string());
-    env.insert("PREM_PYTHON".to_string(), python.clone());
-
-    // Run the bash script
+    println!("🐍 Creating the environment...");
+    let petals_path = get_petals_path(handle);
+    let env = get_environment();
     let _ = Command::new("sh")
         .args([format!("{petals_path}/create_env.sh")])
-        .envs(env)
+        .envs(env.clone())
```

Comment:
```suggestion
        .env_clear()
        .envs(&env)
```

Call env clear on command to ensure nothing gets overwritten. And clone isn't required.

Reply:
@swarnimarun Command is a tauri [Command](https://docs.rs/tauri/latest/tauri/api/process/struct.Command.html) and expect  `HashMap<String, String>`

Reply:
done